src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -1,41 +1,57 @@
 package com.es.phoneshop.web;
 
-import com.es.phoneshop.model.product.Product;
-
+import com.es.phoneshop.model.product.DAO.ArrayListProductDao;
+import com.es.phoneshop.model.product.DAO.ProductDao;
+import jakarta.servlet.ServletConfig;
 import jakarta.servlet.ServletException;
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
+
 import java.io.IOException;
-import java.math.BigDecimal;
-import java.util.ArrayList;
-import java.util.Currency;
-import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
 
 public class ProductListPageServlet extends HttpServlet {
+    public ProductDao productDao;
+
+    @Override
+    public void init(ServletConfig servletConfig) throws ServletException {
+        super.init(servletConfig);
+        productDao = new ArrayListProductDao();
+    }
+
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        request.setAttribute("products", getSampleProducts());
+        request.setAttribute("products", productDao.findProducts());
```

Comment:
in general I suggest to add one more layer - service (for example sigleton ProductService, place it in new package service)
in ProductService we will call ProductDao, throw all possible exceptions
And as a result you can use this service in servlet and you also will be able to catch exceptions from service in servlet

Servlet calls Service -> Service calls Dao
Service is also singleton